lib/helpers/results_helpers.js

```diff
@@ -205,84 +209,117 @@ module.exports = class ResultsHelpers {
+              statementResult.pretty = 'FALSE ([])';
             }
-          } else if (MeasureHelpers.isStatementFunction(library, statementName)) {
+          } else if (MeasureHelpers.isStatementFunction(library, statement_name)) {
             if (doPretty) {
-              statementResults[libraryName][statementName].pretty = 'FUNCTION';
+              statementResult.pretty = 'FUNCTION';
             }
           } else if (doPretty) {
-            statementResults[libraryName][statementName].pretty = `FALSE (${rawStatementResult})`;
+            statementResult.pretty = `FALSE (${rawStatementResult})`;
           }
         }
 
+        if (requestDocument) {
+          const statementResultDocument = new CqmModels.StatementResult(statementResult);
+          statementResults.push(statementResultDocument);
+        } else {
+          statementResults[library_name][statement_name] = statementResult;
+        }
+
         if (includeClauseResults) {
           // create clause results for all localIds in this statement
-          const localIds = MeasureHelpers.findAllLocalIdsInStatementByName(library.elm, statementName);
+          const localIds = MeasureHelpers.findAllLocalIdsInStatementByName(library.elm, statement_name);
           for (const localId in localIds) {
             const clause = localIds[localId];
+            let rawClauseResult = rawClauseResults[library_name] != null
+              ? rawClauseResults[library_name][clause.sourceLocalId != null ? clause.sourceLocalId : localId]
+              : undefined;
+            if (rawClauseResult && rawClauseResult.isMongooseArray) {
+              rawClauseResult = [].concat(rawClauseResult);
+            }
             const clauseResult = {
               // if this clause is an alias or a usage of alias it will get the raw result from the sourceLocalId.
-              raw:
-                rawClauseResults[libraryName] != null
-                  ? rawClauseResults[libraryName][clause.sourceLocalId != null ? clause.sourceLocalId : localId]
-                  : undefined,
-              statementName,
+              raw: rawClauseResult,
+              statement_name,
+              library_name,
+              localId,
             };
+
             clauseResult.final = this.setFinalResults({
               statementRelevance,
-              statementName,
+              statement_name,
               rawClauseResults,
-              libraryName,
+              library_name,
               localId,
               clause,
               rawResult: clauseResult.raw,
             });
-            clauseResults[libraryName][localId] = clauseResult;
+            if (requestDocument) {
+              const clauseResultDocument = new CqmModels.ClauseResult(clauseResult);
+              clauseResults.push(clauseResultDocument);
+            } else {
+              clauseResults[library_name][localId] = clauseResult;
```

Comment:
This is not getting hit. I think you need a test that does both `includeClauseResults = true` and `requestDocument = false`.